ert_gui/tools/plot/storage_client.py

```diff
@@ -99,53 +113,43 @@ def data_for_key(self, case, key):
         """ Returns a pandas DataFrame with the datapoints for a given key for a given case. The row index is
             the realization number, and the column index is a multi-index with (key, index/date)"""
 
-        r = requests.get("{base}/ensembles".format(base=self._BASE_URI))
-
-        ens_pointer = r.json()["ensembles"][0]["ref_url"]
-
-        r = requests.get(ens_pointer)
+        ensembles = ref_request("{base}/ensembles".format(base=self._BASE_URI))
 
-        ens_schema = r.json()
+        ens = [ens for ens in ensembles["ensembles"] if ens["name"]==case][0]
+        ens_schema = ref_request(ens["ref_url"])
 
         df = pd.DataFrame()
-        for real in ens_schema["realizations"]:
-
-            r = requests.get(real["ref_url"])
-            realization = r.json()
-            key_df = pd.DataFrame()
-            for resp in realization["responses"]:
-                if resp["name"] != key:
-                    continue
-
-                r = requests.get(resp["data_url"])
 
-                # TODO: simplified for now, expected structure not in place
-                # Need to add index as well
+        # Response Key
+        for resp in ens_schema["responses"]:
+            response = ref_request(resp["ref_url"])
 
-                # issue:
-                # r.json() -> json.decoder.JSONDecodeError: Extra data: line 1 column 8 (char 7)
-                data = pd.Series(convert_response_to_float(r), name=real["name"])
+            if response["name"] != key:
```

Comment:
this can be checked on the resp in stead to avoid some requests

Reply:
Fixed